lib/hmac.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * FILE:    hmac.c
+ * AUTHORS: Colin Perkins
+ *
+ * HMAC message authentication (RFC2104)
+ *
+ * Copyright (c) 1998-2000 University College London
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, is permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. All advertising materials mentioning features or use of this software
+ *    must display the following acknowledgement:
+ *      This product includes software developed by the Computer Science
+ *      Department at University College London
+ * 4. Neither the name of the University nor of the Department may be used
+ *    to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHORS AND CONTRIBUTORS ``AS IS'' AND
```

Comment:
After doing some Google searches, I found an old Dr. Doob's article (http://www.drdobbs.com/security/the-hmac-algorithm/184410908) providing the listing to an HMAC MD5 algorithm. This one, however, does not have any copyright notice. I change the code to use this new algorithm and I added the 2-clause BSD notice as you suggested.